tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists.py

```diff
@@ -2302,3 +2304,950 @@ def test_bal_cross_block_ripemd160_state_leak(
+        Op.CODECOPY(0, execution_code_size, init_code_size)
+        + Op.SSTORE(
+            0x00,
+            Op.CREATE2(
+                value=endowment,
+                offset=0,
+                size=init_code_size,
+                salt=0,
+            ),
+        )
+        + Op.STOP
+    )
+    # Combine execution code with embedded initcode
+    factory_code = bytes(factory_execution) + init_code_bytes
+
+    factory = pre.deploy_contract(code=factory_code, balance=factory_balance)
+
+    # Calculate the CREATE2 target address
+    created_address = compute_create_address(
+        address=factory,
+        nonce=1,
+        salt=0,
+        initcode=init_code_bytes,
+        opcode=Op.CREATE2,
+    )
+
+    tx = Transaction(
+        sender=alice,
+        to=factory,
+        gas_limit=1_000_000,
+    )
+
+    block = Block(
+        txs=[tx],
+        expected_block_access_list=BlockAccessListExpectation(
+            account_expectations={
+                alice: BalAccountExpectation(
+                    nonce_changes=[BalNonceChange(tx_index=1, post_nonce=1)],
+                ),
+                factory: BalAccountExpectation(
+                    nonce_changes=[BalNonceChange(tx_index=1, post_nonce=2)],
+                    # Balance changes: loses endowment (100)
+                    balance_changes=[
+                        BalBalanceChange(
+                            tx_index=1,
+                            post_balance=factory_balance - endowment,
+                        )
+                    ],
+                ),
+                # Oracle: storage changes for slot 0x01
+                oracle: BalAccountExpectation(
+                    storage_changes=[
+                        BalStorageSlot(
+                            slot=0x01,
+                            slot_changes=[
+                                BalStorageChange(tx_index=1, post_value=0x42)
+                            ],
+                        )
+                    ],
+                ),
+                # Created address: ephemeral (created and destroyed same tx)
+                # - storage_reads for slot 0x01 (aborted write becomes read)
+                # - NO nonce/code/storage/balance changes
+                created_address: BalAccountExpectation(
+                    storage_reads=[0x01],
+                    storage_changes=[],
+                    nonce_changes=[],
+                    code_changes=[],
+                    balance_changes=[],
+                ),
+            }
+        ),
+    )
+
+    blockchain_test(
+        pre=pre,
+        blocks=[block],
+        post={
+            alice: Account(nonce=1),
+            factory: Account(nonce=2, balance=factory_balance - endowment),
+            oracle: Account(storage={0x01: 0x42}),
+            # Created address doesn't exist - destroyed in same tx
+            created_address: Account.NONEXISTENT,
+        },
+    )
+
+
+def test_bal_revert_insufficient_funds(
```

Comment:
We should add `call` to this test name and also move this file to `*_opcodes.py` as mentioned above.